src/scripts/kalimdor/feralas/dire_maul/dreadsteed_ritual.cpp

```diff
@@ -755,16 +757,25 @@ struct go_ritual_nodeAI: public GameObjectAI
             {
                 if (instance_dire_maul* instance = (instance_dire_maul*)me->GetInstanceData())
                 {
-                    if (GameObject* candleAura = instance->GetGameObject(instance->GetData64(GO_RITUAL_CANDLE_AURA)))
+                    if (spell == SPELL_CANDLE_AURA)
                     {
-                        if (Unit* target = instance->GetMap()->GetPlayer(instance->GetData64(DATA_DREADSTEED_RITUAL_PLAYER)))
+                        for (int i = 0; i < MAX_TARGETS; i++)
                         {
-                            if (spell == SPELL_CANDLE_AURA)
-                                candleAura->CastSpell(target, spell, true);
-                            else
-                                me->CastSpell(target, spell, true);
+                            if (Creature* trigger = me->FindNearestCreature(NPC_XOROTHIAN_IMP, 40.0f))
+                            {
+                                if (!trigger->HasAura(spell))
+                                    trigger->AddAura(spell);
+                            }
+
+                            if (Creature* trigger = me->FindNearestCreature(NPC_DREAD_GUARD, 40.0f))
+                            {
+                                if (!trigger->HasAura(spell))
+                                    trigger->AddAura(spell);
+                            }
                         }
```

Comment:
Something of the like:
if (spell == SPELL_CANDLE_AURA) {
    std::list<Creature*> targets;
    GetCreatureListWithEntryInGrid(targets, me, NPC_XOROTHIAN_IMP, 40.0f);
    GetCreatureListWithEntryInGrid(targets, me, NPC_DREAD_GUARD, 40.0f);

    for (auto* candle_target : targets) {
        // Refresh the aura if we need to apply it again.
        // Maybe a refresh aura or something, w/e.
        if (candle_target->HasAuara(spell)) {
            candle_target->RemoveAura(spell);
        }
        candle_target->AddAura(spell);
    }
}
[7:26 PM]
C++11 is not supported, you can replace the inner loop for:
for (list<Creature*>::iterator it = targets.begin(); it != targets.end(); ++it) {
    Creature* target = *it;
}

Reply:
Suggesting this, because this loop seems to always pick the same target.